arangod/IResearch/IResearchRocksDBInvertedIndex.cpp

```diff
@@ -0,0 +1,257 @@
+                                          std::string const& dbname) const {
+  // FIXME: implement!
+  return false;
+}
+
+std::shared_ptr<Index> IResearchRocksDBInvertedIndexFactory::instantiate(LogicalCollection& collection,
+                                                                  velocypack::Slice definition,
+                                                                  IndexId id,
+                                                                  bool isClusterConstructor) const {
+  IResearchViewMeta indexMeta;
+  std::string errField;
+  if (!indexMeta.init(definition, errField)) {
+    LOG_TOPIC("a9ccd", ERR, iresearch::TOPIC)
+      << (errField.empty() ?
+           (std::string("failed to initialize index from definition: ") + definition.toString())
+           : (std::string("failed to initialize index from definition, error in attribute '")
+               + errField + "': " + definition.toString()));
+     return nullptr;
+  }
+  InvertedIndexFieldMeta fieldsMeta;
+  if (!fieldsMeta.init(_server, definition, false, errField,
+                        collection.vocbase().name())) {
+    LOG_TOPIC("18c17", ERR, iresearch::TOPIC)
+      << (errField.empty() ?
+           (std::string("failed to initialize index from definition: ") + definition.toString())
+           : (std::string("failed to initialize index from definition, error in attribute '")
+               + errField + "': " + definition.toString()));
+    return nullptr;
+  }
+  auto nameSlice = definition.get(arangodb::StaticStrings::IndexName);
+  std::string indexName;
+  if (!nameSlice.isNone()) {
+    if (!nameSlice.isString() || nameSlice.getStringLength() == 0) {
+      LOG_TOPIC("91ebd", ERR, iresearch::TOPIC) <<
+        std::string("failed to initialize index from definition, error in attribute '")
+        + arangodb::StaticStrings::IndexName + "': "
+        + definition.toString();
+      return nullptr;
+    }
+    indexName = nameSlice.copyString();
+  }
+
+  auto objectId = basics::VelocyPackHelper::stringUInt64(definition, arangodb::StaticStrings::ObjectId);
+  auto index = std::make_shared<IResearchRocksDBInvertedIndex>(id, collection, objectId, indexName, std::move(fieldsMeta));
+
+  auto initRes  = index->init([this]() -> irs::directory_attributes {
+    auto& selector = _server.getFeature<EngineSelectorFeature>();
+    TRI_ASSERT(selector.isRocksDB());
+    auto& engine = selector.engine<RocksDBEngine>();
+    auto* encryption = engine.encryptionProvider();
+    if (encryption) {
+      return irs::directory_attributes{0, std::make_unique<RocksDBEncryptionProvider>(*encryption,
+                                                          engine.rocksDBOptions())};
+    }
+    return irs::directory_attributes{};
+  });
+
+  if (initRes.fail()) {
+    LOG_TOPIC("9c9ac", ERR, iresearch::TOPIC)
+        << std::string(
+               "failed to do an init for index from definition, error setting "
+               "index properties ")
```

Comment:
Technically, this is not exactly right. The "index properties" are only set afterwards (line 109). So probably we should change the error message here to better reflect what's going on?
```suggestion
        << "failed to do an init for index from definition, error setting index properties: "
```